src/librustdoc/html/render/sorted_json.rs

```diff
@@ -0,0 +1,85 @@
+use itertools::Itertools as _;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::borrow::Borrow;
+use std::fmt;
+
+/// Prerenedered json.
+///
+/// Arrays are sorted by their stringified entries, and objects are sorted by their stringified
+/// keys.
+///
+/// Must use serde_json with the preserve_order feature.
+///
+/// Both the Display and serde_json::to_string implementations write the serialized json
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+#[serde(from = "Value")]
+#[serde(into = "Value")]
+pub(crate) struct SortedJson(String);
+
+impl SortedJson {
+    /// If you pass in an array, it will not be sorted.
+    pub(crate) fn serialize<T: Serialize>(item: T) -> Self {
+        SortedJson(serde_json::to_string(&item).unwrap())
+    }
+
+    /// Serializes and sorts
+    pub(crate) fn array<T: Borrow<SortedJson>, I: IntoIterator<Item = T>>(items: I) -> Self {
+        let items = items
+            .into_iter()
```

Comment:
Sadly I can't do that because of lifetime issues
```
error[E0515]: cannot return value referencing function parameter `x`
  --> src/librustdoc/html/render/sorted_json.rs:28:47
   |
28 |         let items = items.into_iter().map(|x| x.borrow()).sorted_unstable().format(",");
   |                                               -^^^^^^^^^
   |                                               |
   |                                               returns a value referencing data owned by the current function
   |                                               `x` is borrowed here
```